src/lib/mergeThemes.ts

```diff
@@ -56,6 +56,8 @@ export const mergeComponentVariables = (
   const initial = siteVariables => callable(target)(siteVariables)
 
   return sources.reduce<ComponentVariablesPrepared>((acc, next) => {
+    if (!next) return acc
```

Comment:
nit: Before this change `mergeComponentVariables(undefined, undefined)` returned and empty object, now it returns `undefined` - as a result if component has no variables, it receives `variables=undefined` in it's `componentStyles()`. Not sure this is the best practice.

Reply:
This still returns an object because the "initial" value is an object, which is the initial acc value.

I've added a test that asserts this just be sure:
```jsx
  test('always returns an object', () => {
    expect(mergeThemes(undefined, undefined)).toMatchObject({})
    expect(mergeThemes(null, undefined)).toMatchObject({})
    expect(mergeThemes(undefined, null)).toMatchObject({})
    expect(mergeThemes(null, null)).toMatchObject({})

    expect(mergeThemes({}, undefined)).toMatchObject({})
    expect(mergeThemes(undefined, {})).toMatchObject({})
    expect(mergeThemes({}, {})).toMatchObject({})

    expect(mergeThemes({}, null)).toMatchObject({})
    expect(mergeThemes(null, {})).toMatchObject({})
    expect(mergeThemes({}, {})).toMatchObject({})
  })
```

Reply:
That's true for `mergeThemes()`, but not for `mergeComponentVariables()` which is called directly in `renderComponent()`:

```js
expect(mergeComponentVariables(undefined, undefined)()).toMatchObject({})

received value must be an object.
Received: undefined
```

For that reason, Header component fails to render in the branch. (The fact that it uses a variable and doesn't have any headerVariables is another story, will talk to @kuzhelov about that).